File: hardening/iso_date_oracle.rb
# Independent-runtime oracle: reads one date or timestamp with Ruby's Date.parse and prints the calendar
# date it denotes and that date's Julian day number.
# Usage: ruby iso_date_oracle.rb '<date or timestamp>'
require 'date'
day = Date.parse(ARGV[0])
puts day.strftime('%Y-%m-%d')
puts day.jd

File: hardening/iso_date_oracle.php
<?php
// Independent-runtime oracle: reads one date or timestamp with PHP's DateTimeImmutable and prints the
// calendar date it denotes and the instant it denotes as this runtime counts it.
// Usage: php iso_date_oracle.php '<date or timestamp>'
$moment = new DateTimeImmutable($argv[1]);
echo $moment->format('Y-m-d'), "\n";
echo $moment->getTimestamp(), "\n";
